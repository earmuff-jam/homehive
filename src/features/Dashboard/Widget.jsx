import { CSS } from "@dnd-kit/utilities";
import { useSortable } from "@dnd-kit/sortable";

import { CancelRounded, DragIndicatorRounded } from "@mui/icons-material";
import { Badge, Box, IconButton, Paper, Stack, Tooltip } from "@mui/material";

import RowHeader from "src/common/RowHeader/RowHeader";

export default function Widget({ editMode, widget = {}, handleRemoveWidget }) {
  const { attributes, listeners, setNodeRef, transform, transition } =
    useSortable({
      id: widget.widgetID,
    });

  const style = {
    transition,
    transform: CSS.Transform.toString(transform),
  };

  return (
    <Box sx={{ ...style }} ref={setNodeRef}>
      <Badge
        badgeContent={
          editMode && (
            <IconButton
              size="small"
              color="error"
              disableRipple
              disableFocusRipple
              disableTouchRipple
              onClick={() => handleRemoveWidget(widget?.widgetID || "")}
            >
              <CancelRounded fontSize="small" />
            </IconButton>
          )
        }
      >
        <Box
          component={Paper}
          {...widget.config}
          sx={{
            padding: 1,
            overflow: "auto",
            backgroundColor: "background.paper",
          }}
        >
          <Stack direction="row" spacing={1}>
            {editMode && (
              <Tooltip title="Drag and drop to restructure widget layout">
                <IconButton
                  size="small"
                  {...attributes}
                  {...listeners}
                  disableRipple
                  disableFocusRipple
                  disableTouchRipple
                  color="primary"
                  sx={{ cursor: "move" }}
                >
                  <DragIndicatorRounded fontSize="inherit" />
                </IconButton>
              </Tooltip>
            )}
            <RowHeader
              title={widget.label}
              caption={widget.caption}
              sxProps={{
                textAlign: "left",
                fontWeight: "bold",
                color: "text.secondary",
              }}
            />
          </Stack>
        </Box>
      </Badge>
    </Box>
  );
}
