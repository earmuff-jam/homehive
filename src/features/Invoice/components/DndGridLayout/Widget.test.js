import React from "react";
import { render, screen } from "@testing-library/react";
import Widget from "./Widget";

describe("Widget", () => {
  const mockHandleRemoveWidget = jest.fn();

  const widget = {
    widgetID: "1",
    config: {
      width: 200,
      height: 100,
    },
  };

  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(
      <Widget
        editMode={false}
        widget={widget}
        handleRemoveWidget={mockHandleRemoveWidget}
      >
        <div>Child Content</div>
      </Widget>
    );

    expect(screen.getByText(/Child Content/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("shows edit icons when in edit mode", () => {
    const { asFragment } = render(
      <Widget
        editMode={true}
        widget={widget}
        handleRemoveWidget={mockHandleRemoveWidget}
      >
        <div>Child Content</div>
      </Widget>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
