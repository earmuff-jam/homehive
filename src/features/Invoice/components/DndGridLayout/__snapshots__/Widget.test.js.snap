// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <span
      class="MuiBadge-root css-63ezxx-MuiBadge-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiBox-root css-1ka4bzq-MuiPaper-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiStack-root css-niqf4j-MuiStack-root"
        >
          <div>
            Child Content
          </div>
        </div>
      </div>
      <span
        class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular css-2ao5rv-MuiBadge-badge"
      />
    </span>
  </div>
</DocumentFragment>
`;

exports[`Widget shows edit icons when in edit mode 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <span
      class="MuiBadge-root css-63ezxx-MuiBadge-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiBox-root css-1ka4bzq-MuiPaper-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiStack-root css-niqf4j-MuiStack-root"
        >
          <button
            aria-describedby=""
            aria-disabled="false"
            aria-label="Drag and drop to restructure widget layout"
            aria-roledescription="sortable"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-16jdto3-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            role="button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
              data-testid="DragIndicatorRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2m-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2m0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2"
              />
            </svg>
          </button>
          <div>
            Child Content
          </div>
        </div>
      </div>
      <span
        class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular css-2ao5rv-MuiBadge-badge"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeSmall css-ugcj28-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
            data-testid="CancelRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2m4.3 14.3c-.39.39-1.02.39-1.41 0L12 13.41 9.11 16.3c-.39.39-1.02.39-1.41 0a.996.996 0 0 1 0-1.41L10.59 12 7.7 9.11a.996.996 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0L12 10.59l2.89-2.89c.39-.39 1.02-.39 1.41 0s.39 1.02 0 1.41L13.41 12l2.89 2.89c.38.38.38 1.02 0 1.41"
            />
          </svg>
        </button>
      </span>
    </span>
  </div>
</DocumentFragment>
`;
